File: packages/PIPS/validation/Paws/tools/openmp/nowait.c
#include <math.h>

void a92(int n, float *a, float *b, float *c, float *y, float *z) {

	int i;
	
	for (i=0; i<n; i++)
		c[i] = (a[i] + b[i]) / 2.0;

	for (i=0; i<n; i++)
		z[i] = sqrt(c[i]);

	for (i=1; i<=n; i++)
		y[i] = z[i-1] + a[i];
	
}

File: packages/PIPS/validation/Paws/tools/openmp/nowait.f
      SUBROUTINE A92(N, A, B, C, Y, Z)
      INTEGER N
      REAL A(*), B(*), C(*), Y(*), Z(*)
      INTEGER I

      DO I=1,N
            C(I) = (A(I) + B(I)) / 2.0
      ENDDO

      DO I=1,N
            Z(I) = SQRT(C(I))
      ENDDO

      DO I=2,N+1
            Y(I) = Z(I-1) + A(I)
      ENDDO
      END
